### ISSUE TITLE
Provide access to local files - filesystem backend - use web File API

### ISSUE THREAD

**Issue by ieugen** [200ok-ch/organice#921]
Hi,

This has been discussed in the past but I believe a local file access is still very valuable and possible to implement. 
See https://github.com/200ok-ch/organice/issues/93 .

### Context:

I started using org-mode recently and tried organice a few times. 
It hanged and did not perform well - web page was blank. 
I was trying to do grocery shopping at the local flee market.
I can tell you, it was not a nice experience :) . 

I also had to sign in to online service and/or use git - which creates a lot of commits. 

I normally use Syncthing to synchronize files between laptop, servers and mobile phone. https://syncthing.net/ 
It works great, I am very happy with it (stopped using dropbox and gdrive, etc). 
Local storage is much more reliable than internet connection. 

### Proposal:

I believe it should be possible to use organice with local files. 

The web browser has File API https://developer.mozilla.org/en-US/docs/Web/API/File_API that can be used on all major platforms, including mobile devices: https://caniuse.com/?search=file . 

It allows you to open a file and read it's contents. 
There is an example that you can try out on your mobile, on the MDN page above. 
I tried it on my Android 10 and I managed to open a local csv in my case. 
I am sure the same could be used to open local org files and parse them.

There is also a guide on how to access files (and even list directories) https://web.dev/read-files/ . 
It shows different techniques on how to open files.

I will see if I can find some time to check the code out and do POC / PR .  
It would be nice if someone with more organice experience can check if my solution is viable. 

Thanks,
Eugen

Related:
* https://github.com/200ok-ch/organice/issues/663

**Comment by ieugen** [comment]
As I wrote this I realized that there is no JS API to write a file to disk from the browser. 
There is the filesystem API that creates a virtual (isolated FS) but then the problem remains - how to sync to real filesystem https://developer.mozilla.org/en-US/docs/Web/API/File_System_Access_API  

Perhaps the browser could open a  "Save as" dialog to "download" the content to the actual file. 
But I imagine the UX would not be stellar.

Probably a good solution would be to package this web app as a native application and expose the FS API that way.
Has anyone tried doing that ?
